llvm/test/CodeGen/X86/known-bits.ll

```diff
@@ -115,7 +115,7 @@ define i128 @knownbits_mask_addc_shl(i64 %a0, i64 %a1, i64 %a2) nounwind {
 ; X64-NEXT:    andq $-1024, %rdi # imm = 0xFC00
 ; X64-NEXT:    andq $-1024, %rsi # imm = 0xFC00
 ; X64-NEXT:    addq %rdi, %rsi
-; X64-NEXT:    adcq $0, %rdx
+; X64-NEXT:    adcl $0, %edx
```

Comment:
I'm surprised this wasn't already cleaned up by a combine later

Reply:
There's a fshl+add combine that gets triggered now that reduces the size of the add. The uaddo eventually got turned into add, but it was later and something else changed that prevented the fshl+add combine from running at that point.